ingestion/src/metadata/profiler/interface/nosql/profiler_interface.py

```diff
@@ -0,0 +1,207 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+#  pylint: disable=arguments-differ
+
+"""
+Interfaces with database for all database engine
+supporting sqlalchemy abstraction layer
+"""
+import traceback
+from typing import Dict, Iterable, List, Optional, Type
+
+from sqlalchemy import Column
+
+from metadata.generated.schema.entity.data.table import TableData
+from metadata.generated.schema.tests.customMetric import CustomMetric
+from metadata.mixins.pandas.pandas_mixin import PandasInterfaceMixin
+from metadata.profiler.adaptors.factory import factory
+from metadata.profiler.adaptors.nosql_adaptor import NoSQLAdaptor
+from metadata.profiler.api.models import ThreadPoolMetrics
+from metadata.profiler.interface.profiler_interface import ProfilerInterface
+from metadata.profiler.metrics.core import Metric, MetricTypes
+from metadata.profiler.metrics.nosql_metric import NoSQLMetric
+from metadata.profiler.metrics.registry import Metrics
+from metadata.utils.logger import profiler_interface_registry_logger
+from metadata.utils.sqa_like_column import SQALikeColumn
+
+logger = profiler_interface_registry_logger()
+
+
+class NoSQLProfilerInterface(ProfilerInterface, PandasInterfaceMixin):
+    """
+    Interface to interact with registry supporting
+    sqlalchemy.
+    """
+
+    # pylint: disable=too-many-arguments
+
+    def _get_sampler(self):
+        return None
+
+    def _compute_table_metrics(
+        self,
+        metrics: List[Type[Metric and NoSQLMetric]],
+        runner: NoSQLAdaptor,
+        *args,
+        **kwargs,
+    ):
+        result = {}
+        for metric in metrics:
+            try:
+                fn = metric().nosql_fn(runner)
+                result[metric.name()] = fn(self.table)
+            except Exception as exc:
+                logger.debug(
+                    f"{traceback.format_exc()}\n"
+                    f"Error trying to compute metric {metric} for {self.table.fullyQualifiedName}: {exc}"
+                )
+                raise RuntimeError(
```

Comment:
We should not raise in the loop. The idea is that if we fail to compute 1 metric then we should log the error but continue to compute the rest of the metrics so we don't stop the pipeline mid way.